### DISCUSSION TITLE
SEB detects computer as Virtual machine

### DISCUSSION THREAD

**Discussion by ThomasL-AP** [SafeExamBrowser/seb-win-refactoring#787] [Q&A]
Hi, I'm IT support at a school and we have multiple students unable to start their exam with SEB. 
They all get the error (translated): "This computer appears to be a virtual machine. The selected configuration does not allow SEB to be executed in a virtual machine."
I have 6 cases. All got their laptop from the same source: they bought the laptop from their former school (all student come from same school) and kept it after graduation. The laptops should have got a factory reset, as one student told me. I currently can only work on one of them but they should be all the same: HP ProBook 445 G8. The one I have runs Windows 11 22H2

I tried many solutions including:
- disabling all I can find related to virtualisation or Hyper V in Windows components
- reinstalling SEB (version 3.6.0.633)
- disabling virtualisation in BIOS
- installing all windows update.
- removing components in devicemanager related to virtualisation

I'll attach relevant logs and screenshots.
Most relevant/similar discussion I've found online is: https://github.com/SafeExamBrowser/seb-win-refactoring/issues/604 

Could you help me find out why these computers appear to be virtual machines? And also finding a solution.

Thanks in advance.

Excuse me for the dutch screenshots... 
Error notification:
<img width="635" alt="Schermafbeelding 2024-01-17 141604" src="https://github.com/SafeExamBrowser/seb-win-refactoring/assets/156918144/754c5d11-a2a1-4855-a450-b977d389524a">
Windows components:
<img width="960" alt="Schermafbeelding 2024-01-17 142036" src="https://github.com/SafeExamBrowser/seb-win-refactoring/assets/156918144/fe3bf849-549b-44f8-9bd3-b6e276c1ed02">
[2024-01-17_11h09m24s_Browser.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963907/2024-01-17_11h09m24s_Browser.log)
[2024-01-17_11h09m24s_Client.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963908/2024-01-17_11h09m24s_Client.log)
[2024-01-17_11h09m24s_Runtime.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963909/2024-01-17_11h09m24s_Runtime.log)
[cmd.txt](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963906/cmd.txt)

**Comment by dbuechel** [comment]
This might be related to an issue we fixed for the upcoming version 3.7.0, see #747:

```
2024-01-17 11:09:48.295 [06] - INFO: Validating virtual machine policy...
2024-01-17 11:09:48.306 [06] - WARNING: [Registry] Failed to get sub keys for 'HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\TaskFlow\DeviceCache'.
2024-01-17 11:09:48.307 [06] - DEBUG: [VirtualMachineDetector] Computer 'DESKTOP-M4E2IMT' appears to be a virtual machine.
2024-01-17 11:09:48.308 [06] - ERROR: Detected virtual machine while SEB is not allowed to be run in a virtual machine! Aborting...
```

Could you please download the latest beta version of SEB 3.7.0 and check whether that fixes your issue?

**Reply by ThomasL-AP** [reply]
I'm getting the same issue and log file state the same warning/debug/error.

The device registery does not have a sub key "DeviceCache" It ends at. TaskFlow, which only contains a key "UpgradeVersion" | REG_DWORD | 0x00000002 (2)

**Reply by dbuechel** [reply]
@Notselwyn I think we'd need your expertise in this here case, could you please have a look at the issue?

**Reply by Notselwyn** [reply]
The issue is most likely not related to DeviceCache, as the virtual machine detector properly detects and handles the lack of sub keys. The warning message is intended (as part of the Registry class behaviour). It seems there was another reason why the VM detector was triggered. 

@dbuechel Perhaps we could issue a test release which mentions which VM detection feature gets triggered to ease debugging? Or perhaps even put that into the official release in case we need to handle another false flag.

**Reply by dbuechel** [reply]
We unfortunately can't reveal too much information about the VM detection to end users, since that would also help malignant actors in circumventing the measures. But we'll take this up in our internal planning, I think this is an excellent task for you to take on for the upcoming milestone @Notselwyn.

**Comment by dbuechel** [comment]
Please continue the discussion in the issue: #789.

**Comment by horsetek** [comment]
> Hi, I'm IT support at a school and we have multiple students unable to start their exam with SEB. They all get the error (translated): "This computer appears to be a virtual machine. The selected configuration does not allow SEB to be executed in a virtual machine." I have 6 cases. All got their laptop from the same source: they bought the laptop from their former school (all student come from same school) and kept it after graduation. The laptops should have got a factory reset, as one student told me. I currently can only work on one of them but they should be all the same: HP ProBook 445 G8. The one I have runs Windows 11 22H2
> 
> I tried many solutions including:
> 
> * disabling all I can find related to virtualisation or Hyper V in Windows components
> * reinstalling SEB (version 3.6.0.633)
> * disabling virtualisation in BIOS
> * installing all windows update.
> * removing components in devicemanager related to virtualisation
> 
> I'll attach relevant logs and screenshots. Most relevant/similar discussion I've found online is: #604
> 
> Could you help me find out why these computers appear to be virtual machines? And also finding a solution.
> 
> Thanks in advance.
> 
> Excuse me for the dutch screenshots... Error notification: <img alt="Schermafbeelding 2024-01-17 141604" width="635" src="https://private-user-images.githubusercontent.com/156918144/297397452-754c5d11-a2a1-4855-a450-b977d389524a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNTUyMDksIm5iZiI6MTcyNzM1NDkwOSwicGF0aCI6Ii8xNTY5MTgxNDQvMjk3Mzk3NDUyLTc1NGM1ZDExLWEyYTEtNDg1NS1hNDUwLWI5NzdkMzg5NTI0YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNlQxMjQ4MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYTZkMzEyOTMwMDY3NGJjODliNDhjZDFlNjhhZWJhZWRmMGIwNTJjNWZkMWI1YTk0MDVmZDkxZGE2N2YxM2FmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dzCdxZWt8hUrORPsZrqxgK9t2Ogqh9YzTnNh6zIksQw"> Windows components: <img alt="Schermafbeelding 2024-01-17 142036" width="960" src="https://private-user-images.githubusercontent.com/156918144/297397483-fe3bf849-549b-44f8-9bd3-b6e276c1ed02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczNTUyMDksIm5iZiI6MTcyNzM1NDkwOSwicGF0aCI6Ii8xNTY5MTgxNDQvMjk3Mzk3NDgzLWZlM2JmODQ5LTU0OWItNDRmOC05YmQzLWI2ZTI3NmMxZWQwMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNlQxMjQ4MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmNkNWU4YTY1OTlkZTJhZDg3OWRkMzQzYTMyNTliMzE5MDY3MTNjMmE0YmRhMmI2YzQyODdkZmE0ZTlhZmRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MbEGWD6RgWtayUtnPs348EUld2cartaw_U0ALYXR_z0"> [2024-01-17_11h09m24s_Browser.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963907/2024-01-17_11h09m24s_Browser.log) [2024-01-17_11h09m24s_Client.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963908/2024-01-17_11h09m24s_Client.log) [2024-01-17_11h09m24s_Runtime.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963909/2024-01-17_11h09m24s_Runtime.log) [cmd.txt](https://github.com/SafeExamBrowser/seb-win-refactoring/files/13963906/cmd.txt)

**Reply by horsetek** [reply]
I am an IT Support Manager at my school, and I had similar issues with a student laptop not being able to open SEB, "This computer appears to be a virtual machine" always appears when opening the SEB, The issues have been resolved after many troubleshooting. Please feel free to contact me for the details.

**Reply by dbuechel** [reply]
Please try the latest official release version. Many aspects of SEB (especially the VM detection) are constantly being improved and we thus recommend to always use the latest release version.
